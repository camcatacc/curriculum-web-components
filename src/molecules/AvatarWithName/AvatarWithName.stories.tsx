// Modules
import React from "react";

// Storybook Modules
import { ComponentStory, ComponentMeta } from "@storybook/react";

// Elements
import AvatarWithName from "molecules/AvatarWithName/AvatarWithName";
import AvatarImage from "stories/assets/avatar.jpg";

// Template
const Template: ComponentStory<typeof AvatarWithName> = (args) => (
	<div className="w-full md:w-1/4">
		<AvatarWithName {...args} />
	</div>
);

// Variants
export const Standard = Template.bind({});
Standard.args = {
	name: "Johan",
	surname: "Cruyff",
	image: AvatarImage
};

// Export
export default {
	title: "Molecules/AvatarWithName",
	component: AvatarWithName
} as ComponentMeta<typeof AvatarWithName>;
