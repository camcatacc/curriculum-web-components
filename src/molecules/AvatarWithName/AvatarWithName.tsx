// Modules
import React from "react";

// Elements
import AvatarName from "atoms/AvatarName/AvatarName";

// Styles
import * as Styled from "molecules/AvatarWithName/AvatarWithName.styled";

// Definitions
export interface AvatarWithNameProps {
	name: string;
	surname: string;
	image: string;
}

// Element
const AvatarWithName = ({ name, surname, image }: AvatarWithNameProps) => {
	return (
		<>
			<Styled.Container className="w-full relative mb-4">
				<img src={image} alt={name + surname} className="absolute top-0 left-0 bottom-0 right-0 rounded-full" />
			</Styled.Container>
			<AvatarName name={name} surname={surname} />
		</>
	);
};

export default AvatarWithName;
