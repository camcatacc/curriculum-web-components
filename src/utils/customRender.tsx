import { ThemeProvider } from "@material-ui/styles";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { ReactElement } from "react";
import { getMaterialTheme } from "styles/materialUi";

export const customRender = (ui: ReactElement, renderOptions?: RenderOptions & { darkMode?: boolean }): RenderResult => {
	const Wrapper: React.FC = ({ children }) => {
		return <ThemeProvider theme={getMaterialTheme(renderOptions?.darkMode ? "dark" : "light")}>{children}</ThemeProvider>;
	};

	return render(ui, { wrapper: Wrapper, ...renderOptions });
};
