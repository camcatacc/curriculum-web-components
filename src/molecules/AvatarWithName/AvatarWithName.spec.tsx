// Modules
import React from "react";
import { customRender } from "utils/customRender";

// Accessibility
import { axe, toHaveNoViolations } from "jest-axe";

// Elements
import AvatarWithName from "molecules/AvatarWithName/AvatarWithName";

// Images
import AvatarImage from "assets/avatar.jpg";

// Definitions
import type { AvatarWithNameProps } from "molecules/AvatarWithName/AvatarWithName";

// Default Props
const defaultProps: AvatarWithNameProps = {
	name: "David",
	surname: "Camacho Cateura",
	image: AvatarImage
};

// Tests
expect.extend(toHaveNoViolations);
describe("AvatarWithName", () => {
	it("Renders as expected and has no accessibility violations", async () => {
		const { container } = customRender(<AvatarWithName {...defaultProps} />);
		expect(container).toMatchSnapshot();
		expect(await axe(container)).toHaveNoViolations();
	});
});
