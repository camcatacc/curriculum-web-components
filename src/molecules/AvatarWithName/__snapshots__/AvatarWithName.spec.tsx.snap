// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarWithName Renders as expected 1`] = `
<div>
  <div
    class="sc-gsDKAQ mtNEY w-full relative mb-4"
  >
    <img
      alt="DavidCamacho Cateura"
      class="absolute top-0 left-0 bottom-0 right-0 rounded-full"
    />
  </div>
  <h3
    class="MuiTypography-root sc-bdvvtL iytgcZ MuiTypography-h3 MuiTypography-alignCenter"
  >
    David
     
    Camacho Cateura
  </h3>
</div>
`;
